lib/saluki-components/src/destinations/prometheus/mod.rs

```diff
@@ -119,41 +119,46 @@ impl Destination for Prometheus {
                         // Process each metric event in the batch, either merging it with the existing value or
                         // inserting it for the first time.
                         for event in events {
-                            let Event::Metric(metric) = event;
-                            let (context, value, _) = metric.into_parts();
-
-                            // Skip any metric types we can't handle.
-                            let value = match value {
-                                // Skip sets and rates, because we can't convert them to a Prometheus-compatible type.
-                                MetricValue::Rate { .. } | MetricValue::Set { .. } => continue,
-                                // Everything else is fine as-is.
-                                value => value,
-                            };
-
-                            // Generate a Prometheus-specific context, which is essentially the metric name + metric
-                            // type.
-                            //
-                            // We need to do this because we have to group all contexts with the same name to generate
-                            // their payload output as a group, as it's required by the Prometheus exposition specification.
-                            let prom_context = PrometheusMetricContext {
-                                metric_name: context.name().clone(),
-                                metric_type: PrometheusMetricType::from_value(&value),
-                            };
-
-                            let existing_contexts = metrics.entry(prom_context).or_insert_with(IndexMap::new);
-                            match existing_contexts.get_mut(&context) {
-                                Some(existing_value) => {
-                                    existing_value.merge(value);
-                                },
-                                None => {
-                                    if contexts >= CONTEXT_LIMIT {
-                                        debug!("Prometheus destination reached context limit. Skipping metric '{}'.", context.name());
-                                        continue
+                            match event {
+                                Event::Metric(metric)=>{
```

Comment:
Since we only accept metrics here, we can make this cleaner by just doing something like:

```rust
if let Some(metric) = event.into_metric() {
   ...
}|
```

That lets is skip having to even deal with the match expression in the first place.